cmdstanpy/model.py

```diff
@@ -1231,132 +1248,117 @@ def variational(
+                    sampler_args = runset._args.method_args
+                    assert isinstance(
+                        sampler_args, SamplerArgs
+                    )  # make the typechecker happy
+                    sampler_args.fixed_param = True
 
     # pylint: disable=no-self-use
-    def _read_progress(
-        self,
-        proc: subprocess.Popen,  # [] - Popoen is only generic in 3.9
-        pbar: Any,
-        idx: int,
-    ) -> bytes:
-        """
-        Update tqdm progress bars according to CmdStan console progress msgs.
-        Poll process to get CmdStan console outputs,
-        check for output lines that start with 'Iteration: '.
-        NOTE: if CmdStan output messages change, this will break.
-        """
-        pattern = (
-            r'^Iteration\:\s*(\d+)\s*/\s*(\d+)\s*\[\s*\d+%\s*\]\s*\((\S*)\)$'
-        )
-        pattern_compiled = re.compile(pattern, flags=re.IGNORECASE)
-        previous_count = 0
-        stdout = b''
-        changed_description = False  # Changed from 'warmup' to 'sample'
-        pbar.set_description(desc=f'Chain {idx + 1} - warmup', refresh=True)
-
+    def _wrap_sampler_progress_hook(
+        self, chain_id: int, total: int
+    ) -> Optional[Callable[[str], None]]:
+        """Sets up tqdm callback for CmdStan sampler console msgs."""
         try:
-            # iterate while process is sampling
-            while proc.poll() is None and proc.stdout is not None:
-                output = proc.stdout.readline()
-                stdout += output
-                output = output.decode('utf-8').strip()
-                if output.startswith('Iteration'):
-                    match = re.search(pattern_compiled, output)
-                    if match:
-                        current_count = int(match.group(1))
-                        total_count = int(match.group(2))
-
-                        if pbar.total != total_count:
-                            pbar.reset(total=total_count)
-
-                        if (
-                            match.group(3).lower() == 'sampling'
-                            and not changed_description
-                        ):
-                            pbar.set_description(f'Chain {idx + 1} - sample')
-                            changed_description = True
-
-                        pbar.update(current_count - previous_count)
-                        previous_count = current_count
-
-            pbar.set_description(f'Chain {idx + 1} -   done', refresh=True)
-
-            if 'notebook' in type(pbar).__name__:
-                # In Jupyter make the bar green by closing it
-                pbar.close()
-
-        except Exception as e:  # pylint: disable=broad-except
-            get_logger().warning(
-                'Chain %s: Failed to read the progress on the fly. Error: %s',
-                idx,
-                repr(e),
+            from tqdm.auto import tqdm  # type:ignore
+
+            pbar: Any = tqdm(
```

Comment:
```suggestion
            pbar: tqdm = tqdm(
```

I don't think this needs to be Any, we can give it the class annotation

Reply:
fixed